asteroid/utils/torch_utils.py

```diff
@@ -123,3 +124,87 @@ def are_models_equal(model1, model2):
         if p1.data.ne(p2.data).sum() > 0:
             return False
     return True
+
+class OverlapAddWrapper(torch.nn.Module):
+    def __init__(self, nnet, sources, window_size, device, window="hanning", reorder_chunks=True):
+        super(OverlapAddWrapper, self).__init__()
+
+        assert window_size % 2 == 0, "Window size must be even"
+
+        self.to(device)
+        self.device = device
+        if isinstance(nnet, torch.nn.Module):
+            nnet = nnet.to(device)
+        self.function = nnet
+        self.window_size = window_size
+        self.sources = sources
+
+        window = get_window(window, self.window_size).astype("float32")
+        window = torch.from_numpy(window)
+        self.register_buffer("window", window)
+        self.reorder_chunks = reorder_chunks
+
+    def _reorder_sources(self, current, previous):
+        # we compute xcorr for all permutations
+        # we get index for min
+        batch, frames = current.size()
+        current = current.reshape(-1, self.sources, frames)
+        previous = previous.reshape(-1, self.sources, frames)
+
+        pw_losses = PITLossWrapper.get_pw_losses(lambda x, y: torch.sum((x.unsqueeze(1) * y.unsqueeze(2))),
+                                                 current[..., :self.window_size//2],
+                                                 previous[..., -self.window_size//2:])
+        _, perms = PITLossWrapper.find_best_perm(pw_losses, self.sources)
+        current = PITLossWrapper.reorder_source(current, self.sources, perms)
+        return current.reshape(batch, frames)
+
+    def forward(self, x):
+        with torch.no_grad():
+            x = x.to(self.device)
+            assert len(x.shape) == 3
+
+            # Overlap and add:
+            # [batch, channels, n_frames] -> [batch, channels, window_size, n_chunks]
+            batch, channels, n_frames = x.size()
+            hop_size = self.window_size // 2
```

Comment:
Do we really want to hard code this? 
It would be cool to be able to reproduce table 5 from [this paper](https://arxiv.org/pdf/2003.01531.pdf). It wouldn't be too complicated with this class right?

Reply:
IDK i usually use hop_size//2 because we get perfect reconstruction with hamming window

Reply:
Yes, true. 
Other question then, is the window needed? 
Alright, let's keep it like this for now, but register `self.hop_size` in the `__init__`  instead of having it hardcoded in two places in the code. If we want to change in the future, we'll be able.

Reply:
I ll add an option for no window